File: exercise/php/day08/src/Domain/Toy.php
<?php

declare(strict_types = 1);

namespace Domain;

class Toy
{
	public const UNASSIGNED = 'UNASSIGNED';

	public const IN_PRODUCTION = 'IN_PRODUCTION';

	public const COMPLETED = 'COMPLETED';

	private string $name;

	private string $state;

	public function __construct(string $name, string $state)
	{
		$this->name = $name;
		$this->state = $state;
	}

	public function getName(): string
	{
		return $this->name;
	}

	public function getState(): string
	{
		return $this->state;
	}

	public function setState(string $state): void
	{
		$this->state = $state;
	}
}

File: exercise/java/day08/src/main/java/domain/Toy.java
package domain;

public class Toy {
    public enum State {
        UNASSIGNED, IN_PRODUCTION, COMPLETED
    }

    private final String name;
    private State state;

    public Toy(String name, State state) {
        this.name = name;
        this.state = state;
    }

    public String getName() {
        return name;
    }

    public State getState() {
        return state;
    }

    public void setState(State state) {
        this.state = state;
    }
}